rotkehlchen/chain/evm/decoding/odos/v2/decoder.py

```diff
@@ -0,0 +1,228 @@
+
+            elif hex_or_bytes_to_address(tx_log.topics[1]) == self.router_address:
+                router_holding_amount[asset] -= FVal(hex_or_bytes_to_int(tx_log.data)) / FVal(10 ** asset.get_decimals())  # noqa: E501
+
+        if self.native_currency.identifier in output_tokens:
+            try:
+                internal_native_in = self.evm_txns.get_and_ensure_internal_txns_of_parent_in_db(
+                    tx_hash=context.transaction.tx_hash,
+                    to_address=self.router_address,
+                    user_address=decoded_data[0],
+                )
+                internal_native_out = self.evm_txns.get_and_ensure_internal_txns_of_parent_in_db(
+                    tx_hash=context.transaction.tx_hash,
+                    from_address=self.router_address,
+                    to_address=decoded_data[0],
+                    user_address=decoded_data[0],
+                )
+            except RemoteError as e:
+                log.error(f'Failed to get internal transactions for odos v2 {self.evm_inquirer.chain_name} swap {context.transaction.tx_hash.hex()} due to {e!s}')  # noqa: E501
+            else:
+                if len(internal_native_in) > 0:
+                    router_holding_amount[self.native_currency] += FVal(internal_native_in[0].value) / FVal(10 ** self.native_currency_decimals)  # noqa: E501
+                if len(internal_native_out) > 0:
+                    router_holding_amount[self.native_currency] -= FVal(internal_native_out[0].value) / FVal(10 ** self.native_currency_decimals)  # noqa: E501
+
+        for held_asset, amount in router_holding_amount.items():
+            if amount != ZERO:
+                fee_events.append(self.base.make_event_next_index(
+                    tx_hash=context.transaction.tx_hash,
+                    timestamp=context.transaction.timestamp,
+                    event_type=HistoryEventType.SPEND,
+                    event_subtype=HistoryEventSubType.FEE,
+                    asset=held_asset,
+                    balance=Balance(amount=amount),
+                    location_label=decoded_data[0],
+                    notes=f'Spend {amount} {held_asset.symbol_or_name()} as an Odos v1 fee',
+                    counterparty=CPT_ODOS_V2,
+                    address=self.router_address,
+                ))
+
+        context.decoded_events.sort(key=lambda event: event.event_subtype.value)
+        context.decoded_events.extend(fee_events)
+        maybe_reshuffle_events(
+            ordered_events=context.decoded_events,
+            events_list=out_events + in_events + fee_events,
+        )
+        return DEFAULT_DECODING_OUTPUT
+
+    # -- DecoderInterface methods
+
+    def addresses_to_decoders(self) -> dict[ChecksumEvmAddress, tuple[Any, ...]]:
+        return {
+            self.router_address: (self._decode_swap,),
+        }
```

Comment:
fits in one line 
![image](https://github.com/user-attachments/assets/46894617-02c6-4ff5-ad3c-81cf1f8350bf)